client/network/src/lib.rs

```diff
@@ -89,14 +89,13 @@ impl BlockAnnounceData {
 	///
 	/// This will not check the signature, for this you should use [`BlockAnnounceData::check_signature`].
 	fn validate(&self, encoded_header: Vec<u8>) -> Result<(), Validation> {
-		let candidate_hash = if let CompactStatement::Seconded(h) =
-			self.statement.unchecked_payload()
-		{
-			h
-		} else {
-			tracing::debug!(target: LOG_TARGET, "`CompactStatement` isn't the candidate variant!",);
-			return Err(Validation::Failure { disconnect: true })
-		};
+		let candidate_hash =
```

Comment:
this was caused with new nightly `2023-05-03`,
I have found this also in my [PR](https://github.com/paritytech/cumulus/pull/2458/files#diff-0fe3ce0d7087ba7f979d9065d8d812b23562d2f4ed5341719aa8480dc48c74cd), which will go to the master soon